neuralcompression/layers/_rate_distortion_loss.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+from typing import List, NamedTuple
+
+import torch
+from torch import Tensor
+from torch.nn import Module, MSELoss
+
+
+class _ForwardReturnType(NamedTuple):
+    bpp: float
+    mse: float
+    rate_distortion: float
+
+
+class RateDistortionLoss(Module):
+    """Rate-distortion loss.
+
+    The rate-distortion loss is the minimum transmission bit-rate for a
+    required quality. It can be obtained without consideration of a specific
+    coding method.
+
+    Args:
+        _lambda: `_lambda = 1` is the solution where the `(rate, distortion)`
+            pair minimizes `rate + distortion`. Increasing `_lambda` will
+            penalize the distortion term so more bits are spent.
```

Comment:
This doesn't need to be private does it?

Reply:
`lambda` is reserved. I went ahead and renamed `lambda` to `trade_off`.